juicebox_scripts/juicebox_assembly_purger.py

```diff
@@ -0,0 +1,131 @@
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see https://www.gnu.org/licenses/agpl-3.0.en.html
+'''
+
+from __future__ import print_function
+import argparse
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("input_assembly")
+    parser.add_argument("output_assembly")
+    parser.add_argument("--exclude_contigs",
+                        default=None,
+                        nargs="+",
+                        help="Names of contigs to exclude")
+    parser.add_argument("--exclude_file",
+                        default=None,
+                        help="Path to file of contigs to exclude (with contig names in "
+                             "first whitespace-delimited column, one per line)")
+    args = parser.parse_args()
+
+    if args.exclude_contigs is None and args.exclude_file is None:
+        raise ValueError("Error: exclude_contigs or exclude_file must be specified")
+
+    return vars(args)
+
+def get_exclude(contigs, file):
+    """Takes a list and/or file of contig names and returns a set of contig names to exclude.
+    Assumes contigs and file are not both None.
+    """
+    exclude = set([])
+    if contigs is not None:
+        for name in contigs:
+            exclude.add(name)
+    if file is not None:
+        with open(file, 'r') as input:
+            for line in input:
+                name = line.split()[0]
+                exclude.add(name)
+
+    return exclude
+
+def filter_assembly(exclude, input_assembly, output_assembly, **kwargs):
+    """Take an input assembly file and create an output assembly file
+    without the contigs specified in the exclude set.
+    """
+    with open(input_assembly, "r") as input, \
+         open(output_assembly, "w") as output:
+        index = 1
+        purged_names = set([])
+        purged_indices = set([])
+        purged_from_scaffolds = set([])
+        for line in input:
+            if line.startswith(">"):
+                name, num, length = line.strip().split()
+                name = name.replace(">", "")
+                if name in exclude:
+                    purged_names.add(name)
+                    purged_indices.add(num)
+                    continue
+                else:
+                    outstring = ">{} {} {}\n".format(name, index, length)
+                    output.write(outstring)
+                    index += 1
+            else:
```

Comment:
wait... does this account for the index frameshifting of the contigs? i would expect the loop above to include a mapping from old to new indices, and to replace the old indices with new (or exclude if they're the excluded indices).

Reply:
if i understand the code above, you are generating new contig-index mappings, but you aren't updating the indices in the scaffolds to match the new indices. should the scaffold indices not reflect the new indices?

i'd like to get this clarified before a merge.

Reply:
Ah, yep, it's accounted for in the header but not in the scaffolds.

Reply:
cool, this makes sense to me. would be good to have a test case that makes sure this is handled properly...

Reply:
Added a bunch of tests.